meilisearch-schema/src/schema.rs

```diff
@@ -101,249 +62,310 @@ impl Schema {
-        };
-        self.indexed_map.clear();
-        for name in data {
-            self.set_indexed(name.as_ref())?;
+    pub fn update_searchable<S: AsRef<str>>(&mut self, data: Vec<S>) -> SResult<()> {
+        let mut searchable = Vec::with_capacity(data.len());
+        for (pos, name) in data.iter().enumerate() {
+            let id = self.insert(name.as_ref())?;
+            self.indexed_position.insert(id, IndexedPos(pos as u16));
+            searchable.push(id);
         }
+        self.searchable.replace(searchable);
         Ok(())
     }
 
-    pub fn set_all_fields_as_indexed(&mut self) {
-        self.indexed = OptionAll::All;
-        self.indexed_map.clear();
+    pub fn set_all_searchable(&mut self) {
+        self.searchable.take();
+    }
 
-        for (_name, id) in self.fields_map.iter() {
-            let pos = self.indexed_map.len() as u16;
-            self.indexed_map.insert(*id, pos.into());
-        }
+    pub fn set_all_displayed(&mut self) {
+        self.displayed.take();
     }
+}
 
-    pub fn set_all_fields_as_displayed(&mut self) {
-        self.displayed = OptionAll::All
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_with_primary_key() {
+        let schema = Schema::with_primary_key("test");
+        assert_eq!(
+            format!("{:?}", schema),
+            r##"Schema { fields_map: FieldsMap { name_map: {"test": FieldId(0)}, id_map: {FieldId(0): "test"}, next_id: FieldId(1) }, primary_key: Some(FieldId(0)), ranked: {}, displayed: None, searchable: None, indexed_position: PositionMap { pos_to_field: [FieldId(0)], field_to_pos: {FieldId(0): IndexedPos(0)} } }"##
+        );
+    }
+
+    #[test]
+    fn primary_key() {
+        let schema = Schema::with_primary_key("test");
+        assert_eq!(schema.primary_key(), Some("test"));
+    }
+
+    #[test]
+    fn test_insert_with_position_base() {
+        let mut schema = Schema::default();
+        let (id, position) = schema.insert_with_position("foo").unwrap();
+        assert!(schema.searchable.is_none());
+        assert!(schema.displayed.is_none());
+        assert_eq!(id, 0.into());
+        assert_eq!(position, 0.into());
+        let (id, position) = schema.insert_with_position("bar").unwrap();
+        assert_eq!(id, 1.into());
+        assert_eq!(position, 1.into());
+    }
+
+    #[test]
+    fn test_insert_with_position_primary_key() {
+        let mut schema = Schema::with_primary_key("test");
+        let (id, position) = schema.insert_with_position("foo").unwrap();
+        assert!(schema.searchable.is_none());
+        assert!(schema.displayed.is_none());
+        assert_eq!(id, 1.into());
+        assert_eq!(position, 1.into());
+        let (id, position) = schema.insert_with_position("test").unwrap();
+        assert_eq!(id, 0.into());
+        assert_eq!(position, 0.into());
+    }
+
+    #[test]
+    fn test_insert_with_position_non_all_searchable_attributes() {}
```

Comment:
whoops
```suggestion
```

Reply:
You forgot to also remove the `#[test]` line and line above that!